### PR TITLE
Add documentation for Iceberg Glue catalog

### PR DESCRIPTION
**Pull request by alexjo2144**
## Description

Adds documentation for configuration of https://github.com/trinodb/trino/pull/10845

> Is this change a fix, improvement, new feature, refactoring, or other?

Documentation

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Documentation

> How would you describe this change to a non-technical end user or system administrator?

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by alexjo2144** [comment]
Adding the WIP label until the code changes are actually merged.

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 1]
```diff
@@ -37,21 +37,50 @@ To use Iceberg, you need:
...
+* Access to a Hive metastore service (HMS) or AWS Glue.
 * Network access from the Trino coordinator to the HMS. Hive
   metastore access with the Thrift protocol defaults to using port 9083.
 
 Configuration
 -------------
 
-Iceberg supports the same metastore configuration properties as the Hive connector.
-At a minimum, ``hive.metastore.uri`` must be configured:
+The connector supports two Iceberg catalog types, you may use either a Hive
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:47]: We never talk about "catalog types" anywhere.  Just say that two different metastores are supported.

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 2]
```diff
@@ -37,21 +37,50 @@ To use Iceberg, you need:
...
+metastore service (HMS) or AWS Glue. The catalog type is determined by the
+``iceberg.catalog.type`` property, it can be set to either ``HIVE_METASTORE``
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.rst:49]: Thanks for the info.

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 3]
```diff
@@ -37,21 +37,50 @@ To use Iceberg, you need:
...
+configuration properties as the Hive connector's Glue setup. See
+:ref:`AWS Glue metastore configuration<hive-glue-metastore>`.
+
+.. code-block:: text
+
+    connector.name=iceberg
+    iceberg.catalog.type=glue
+
+
+Generic Configuration
```
Comment (findepi) [inline docs/src/main/sphinx/connector/iceberg.rst:78]: I don't think we should use the word "Generic" here.

Comment (alexjo2144) [reply docs/src/main/sphinx/connector/iceberg.rst:78]: "Other"?

Comment (alexjo2144) [reply docs/src/main/sphinx/connector/iceberg.rst:78]: "Universal"?

Comment (findepi) [reply docs/src/main/sphinx/connector/iceberg.rst:78]: General? 
or "iceberg configuration properties" like at https://trino.io/docs/current/connector/hive.html#hive-configuration-properties

(btw we seem to be using "Sentence case" and not "Upper Case" for section headers)

Comment (alexjo2144) [reply docs/src/main/sphinx/connector/iceberg.rst:78]: I like "General configuration"

And yes, will fix capitalization, thanks

**Comment by alexjo2144** [comment]
Updated, and removed the WIP label since the code PR was merged.

**Review by findepi — APPROVED** [review]

**Pull request merged**